packages/db/src/query/user.ts

```diff
@@ -235,39 +237,24 @@ export async function setUserAdmin(id: string, isAdmin: boolean) {
     return db.update(user).set({ isAdmin }).where(eq(user.id, id));
 }
 
-/**
- * Get users with due report to create and send reports </br>
- * the report is due if the current date is greater than the last report date + interval or </br>
- * if the current date is equal to the last report date + interval and the current time is greater than the report time </br>
- *
- * @returns The users with due report
- */
-export async function getUsersWitDueReport() {
-    return db
-        .select({ userId: user.id, userName: user.username, email: user.email, receiveMails: reports.receiveMails })
-        .from(reports)
-        .innerJoin(user, eq(user.id, reports.userId))
-        .where(
-            or(
-                gt(sql`DATEDIFF(NOW(), reports.timestamp_last)`, reports.interval),
-                and(
-                    eq(sql`DATEDIFF(NOW(), reports.timestamp_last)`, reports.interval),
-                    lte(reports.time, new Date().getHours()),
-                ),
-            ),
-        );
-}
+export async function createToken(userId: string, type: TokenType = TokenType.Report) {
+    return db.transaction(async (trx) => {
+        await trx.insert(token).values({ userId });
 
-export async function getReportDataByUserId(id: string) {
-    const data = await db.select().from(reports).where(eq(reports.userId, id));
+        const createdToken = await trx
+            .select()
+            .from(token)
+            .where(and(eq(token.userId, userId)))
+            .orderBy(desc(token.createdTimestamp))
+            .limit(1);
+        return createdToken[0].token;
+    });
+}
 
+export async function getUserIdByToken(givenToken: string) {
+    const data = await db.select().from(token).where(eq(token.token, givenToken));
```

Comment:
Hier müssten wir spätestesns in Hinblick auf die Altert-Mails noch mit dem TokenType arbeiten

Reply:
Joa, würde ich aber tatsächlich aber dann erst anpassen